pkg/trace/operation.go

```diff
@@ -124,32 +130,79 @@ func (o *Operation) ID() string {
 }
 
 func (o *Operation) Infof(format string, args ...interface{}) {
-	Logger.Infof("%s: %s", o.header(), fmt.Sprintf(format, args...))
+	msg := fmt.Sprintf(format, args...)
+
+	Logger.Infof("%s: %s", o.header(), msg)
+
+	if o.Logger != nil {
+		o.Logger.Info(msg)
+	}
+}
+
+func (o *Operation) Info(args ...interface{}) {
+	o.Infof(fmt.Sprint(args...))
```

Comment:
May make sense to reverse the logic here between `Infof` and `Info` - that way we're not calling `Sprintf` on a string that it has to parse but instead calling Sprint on a single string.